packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -641,21 +642,22 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     /**
      * Load the stores from a snapshot and returns the runtime.
      * @param context - Context of the container.
-     * @param registryEntries - Mapping to the stores.
-     * @param requestHandler - Request handlers for the container runtime
-     * @param runtimeOptions - Additional options to be passed to the runtime
-     * @param existing - (optional) When loading from an existing snapshot. Precedes context.existing if provided
-     * @param containerRuntimeCtor - (optional) Constructor to use to create the ContainerRuntime instance. This
-     * allows mixin classes to leverage this method to define their own async initializer.
-     */
-    public static async load(
+     * @param containerRuntimeCtor - (optional) Constructor to use to create the ContainerRuntime instance.
+     * This allows mixin classes to leverage this method to define their own async initializer.
+     * @param existing - Pass 'true' if loading from an existing snapshot.
+     * @param registryEntries - Mapping from data store types to their corresponding factories,
+     * @param containerScop - Scope
+     * @param initializeEntryPoint - Promise that resolves to an object which will act as entryPoint for the Container.
+     * This object should provide all the functionality that the Container is expected to provide to the loader layer.
+     */
+    public static async newLoad(
         context: IContainerContext,
-        registryEntries: NamedFluidDataStoreRegistryEntries,
-        requestHandler?: (request: IRequest, runtime: IContainerRuntime) => Promise<IResponse>,
+        containerRuntimeCtor: typeof ContainerRuntime = ContainerRuntime,
         runtimeOptions: IContainerRuntimeOptions = {},
+        existing: boolean,
+        registryEntries: NamedFluidDataStoreRegistryEntries,
         containerScope: FluidObject = context.scope,
-        existing?: boolean,
-        containerRuntimeCtor: typeof ContainerRuntime = ContainerRuntime
+        initializeEntryPoint: (containerRuntime: IContainerRuntime) => Promise<FluidObject>
```

Comment:
i would put this above all opptional params

Reply:
Makes sense, I'll move it.